src/services/extra_data.py

```diff
@@ -57,15 +57,21 @@ def __init__(self, w3: Web3):
         self.w3 = w3
 
     def collect(self, blockstamp: BlockStamp, exited_validators: list[LidoKey], stucked_validators: list[LidoKey]) -> bytes:
-        operators_response = self.w3.kac.get_operators(blockstamp)
-        # TODO get from oracleReportSanityChecker contract
-        max_extra_data_list_items_count = 100
-        stucked_payloads = self.build_validators_payloads(stucked_validators, max_extra_data_list_items_count)
-        exited_payloads = self.build_validators_payloads(exited_validators, max_extra_data_list_items_count)
+        modules = [item.module for item in self.w3.kac.get_operators(blockstamp)]
+        max_items_count = self._get_oracle_report_limits().maxAccountingExtraDataListItemsCount
+        stucked_payloads = self.build_validators_payloads(stucked_validators, modules, max_items_count)
+        exited_payloads = self.build_validators_payloads(exited_validators, modules, max_items_count)
 
         extra_data = self.build_extra_data(stucked_payloads, exited_payloads)
         return self.to_bytes(extra_data)
 
+    def _get_oracle_report_limits(self):
+        abi = next(filter(lambda x: x.get('name') == 'getOracleReportLimits',
+                          self.w3.lido_contracts.oracleReportSanityChecker.abi))
+        output_names = list(map(itemgetter('name'), abi['outputs'][0]['components']))
+        result = self.w3.lido_contracts.oracleReportSanityChecker.functions.getOracleReportLimits().call()
+        return OracleReportLimits(**dict(zip(output_names, result)))
```

Comment:
cool idea
Is it possible to write a function that returns abi to write out separately so that it can be used everywhere?

Reply:
+ We need at least one positive test for this feature

Reply:
Yea will do it
I expect that the next web3py version will be able to do it out of the box: https://github.com/ethereum/web3.py/pull/2799.

Reply:
done